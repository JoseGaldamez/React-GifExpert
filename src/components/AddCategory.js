import React, { useState } from 'react';
import PropTypes from 'prop-types';


const AddCategory = ({setCategories}) => {
    
    const [inputValue, setinputValue] = useState('');
    const _handleInputChange = (e) => {
        setinputValue(e.target.value);
    }
    
    const _handleSubmit = (e) => {
         
        e.preventDefault();
        
        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
            setinputValue('');
        }
        
        
    }
    return (
        <form onSubmit={_handleSubmit}>
            <input placeholder="Buscar..." type="text" value={inputValue} onChange={_handleInputChange} />
        </ form>
    );
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}


export default AddCategory;
