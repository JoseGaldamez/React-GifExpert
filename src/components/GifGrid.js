import React from 'react';
import useFetchGifs from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    
    const { data:images, loading } = useFetchGifs(category);
    
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            
            { loading && <p className="animate__animated animate__flash">Cargando...</p> }
            
        <div className="card-grid">
            { 
            images.length != 0
                ? images.map((img) => {
                        return ( <GifGridItem key={img.id} {...img} /> )
                    })
                : !loading && <p>No se encontraron gifs con este t√≠tulo</p>
           }
        </div>
        </>
    )
}
