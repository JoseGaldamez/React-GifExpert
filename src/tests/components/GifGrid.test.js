import React from "react";
import { shallow } from "enzyme"

import '@testing-library/jest-dom';

import {GifGrid} from '../../components/GifGrid';
import useFetchGifs from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs")

describe("Probando GifGrid", ()=> {
    
    
    test("Analizando Snapshot", ()=> {
        const gifs =  [{
            id: "ABC",
            url: "https://cualquiercos.com/algo.png",
            title: "Cualquier cosa"
        }];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true,
        });
        const wrapper = shallow(<GifGrid category="Hola" />);

        expect(wrapper).toMatchSnapshot();
    });




    
    test("Debe mostrar items cuando se cargan imagenes", () => {
        const gifs =  [{
            id: "ABC",
            url: "https://cualquiercos.com/algo.png",
            title: "Cualquier cosa"
        }, {
            id: "ABCD",
            url: "https://cualquiercos.com/algo.png",
            title: "Cualquier cosa"
        }];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow(<GifGrid category="Hola" />);

        expect( wrapper.find('#loading').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);

    });
})