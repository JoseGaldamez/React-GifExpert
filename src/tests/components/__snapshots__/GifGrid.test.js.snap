// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GifGrid Analizando Snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Hola
  </h3>
  <p
    className="animate__animated animate__flash"
    id="loading"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://cualquiercos.com/algo.png"
    />
  </div>
</Fragment>
`;
